common/oauthTokenManager.go

```diff
@@ -523,61 +500,35 @@ func getAuthorityURL(tenantID, activeDirectoryEndpoint string) (*url.URL, error)
 	return u.Parse(tenantID)
 }
 
-const minimumTokenValidDuration = time.Minute * 5
 type TokenStoreCredential struct {
 	token *azcore.AccessToken
-	lock  sync.RWMutex
 }
 
 // globalTokenStoreCredential is created to make sure that all
 // service clients share same cred object. This is required so that
 // we do not make repeated GetToken calls.
 // This is a temporary fix for issue where we would request a
 // new token from Stg Exp even while they've not yet populated the
-// tokenstore. 
+// tokenstore.
 //
 // This is okay because we use same credential on both source and
 // destination. If we move to a case where the credentials are
 // different, this should be removed.
 //
 // We should move to a method where the token is always read  from
 // tokenstore, and azcopy is invoked after tokenstore is populated.
-//
 var globalTokenStoreCredential *TokenStoreCredential
 var globalTsc sync.Once
 
 func (tsc *TokenStoreCredential) GetToken(_ context.Context, _ policy.TokenRequestOptions) (azcore.AccessToken, error) {
-	// if the token we've has not expired, return the same.
-	tsc.lock.RLock()
-	if time.Until(tsc.token.ExpiresOn) > minimumTokenValidDuration {
-		return *tsc.token, nil
-	}
-	tsc.lock.RUnlock()
-
-	tsc.lock.Lock()
-	defer tsc.lock.Unlock()
-	hasToken, err := tokenStoreCredCache.HasCachedToken()
-	if err != nil || !hasToken {
-		return azcore.AccessToken{}, fmt.Errorf("no cached token found in Token Store Mode(SE), %v", err)
-	}
-
-	tokenInfo, err := tokenStoreCredCache.LoadToken()
-	if err != nil {
-		return azcore.AccessToken{}, fmt.Errorf("get cached token failed in Token Store Mode(SE), %v", err)
-	}
-
-	tsc.token = &azcore.AccessToken{
-		Token:     tokenInfo.AccessToken,
-		ExpiresOn: tokenInfo.Expires(),
-	}
-
+	tsc.token = &azcore.AccessToken{}
```

Comment:
Why are we returning an empty token?

Reply:
I need some context here - lets discuss offline on this.